internal/wasm/encoding/reader.go

```diff
@@ -804,6 +828,6 @@ func readBlockValueType(r io.Reader, v *types.ValueType) error {
 
 func readByte(r io.Reader) (byte, error) {
 	buf := make([]byte, 1)
-	_, err := r.Read(buf)
+	_, err := io.ReadFull(r, buf)
```

Comment:
From my reading of [`Reader`](https://golang.org/pkg/io/#Reader), it could be possible that `r.Read` reads 0 bytes? 🤔 I think we're on the safe side with [`io.ReadFull`](https://golang.org/pkg/io/#ReadFull) here.